src/configure/configure.ts

```diff
@@ -245,7 +292,15 @@ class PipelineConfigurer {
                 title: utils.format(Messages.creatingAzureServiceConnection, this.inputs.targetResource.subscriptionId)
             },
             () => {
-                return this.serviceConnectionHelper.createAzureServiceConnection(this.inputs.targetResource.resource.name, this.inputs.azureSession.tenantId, this.inputs.targetResource.subscriptionId);
+                let scope = this.inputs.targetResource.resource.id;
+                let aadAppName = GraphHelper.generateAadApplicationName(this.inputs.organizationName, this.inputs.projectName);
+                return GraphHelper.createSpnAndAssignRole(this.inputs.azureSession, aadAppName, scope)
+                .then((aadApp) => {
+                    let uniqueNamePostfix = uuid();
```

Comment:
As we are using `uniqueNamePostfix` at multiple places, should we use the same `uniqueNamePostfix` everywhere. So that all resources of one pipeline will follow the same `uniqueNamePostfix`. 
It can be generated as a private property everytime.

Reply:
Good idea ! we can use the same unique postfix for all the resources which require it. Will do it.